modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/DatabaseRegistry.java

```diff
@@ -261,6 +262,21 @@ void retrieveAndUpdateDatabase(String databaseName, GeoIpTaskState.Metadata meta
                 decompress(databaseTmpGzFile, databaseTmpFile);
 
                 Path databaseFile = geoipTmpDirectory.resolve(databaseName);
+                try (TarInputStream is = new TarInputStream(new BufferedInputStream(Files.newInputStream(databaseTmpFile)))) {
```

Comment:
maybe leave a comment here about the structure of the tar file and how it is going to be extracted?

Reply:
Added

Reply:
Added